MCLauncher/MainWindow.xaml.cs

```diff
@@ -52,6 +53,22 @@ public partial class MainWindow : Window, ICommonVersionCommands {
             });
         }
 
+        private async void Button_Click(object sender, RoutedEventArgs e)
+        {
+            Microsoft.Win32.OpenFileDialog openFileDlg = new Microsoft.Win32.OpenFileDialog();
+            Nullable<bool> result = openFileDlg.ShowDialog();
+            if (result == true)
+            {
+                string directory = openFileDlg.SafeFileName + " (imported)";
```

Comment:
Is there a reason to add (imported) to the actual directory name?
I think that this shouldn't be appended in the dirname, but instead just appended in the display layer.